api/v1alpha1/rollingupgrade_types.go

```diff
@@ -117,9 +116,21 @@ const (
 )
 
 var (
-	FiniteStates        = []string{StatusComplete, StatusError}
-	AllowedStrategyType = []string{string(RandomUpdateStrategy), string(UniformAcrossAzUpdateStrategy)}
-	AllowedStrategyMode = []string{string(UpdateStrategyModeLazy), string(UpdateStrategyModeEager)}
+	FiniteStates                = []string{StatusComplete, StatusError}
+	AllowedStrategyType         = []string{string(RandomUpdateStrategy), string(UniformAcrossAzUpdateStrategy)}
+	AllowedStrategyMode         = []string{string(UpdateStrategyModeLazy), string(UpdateStrategyModeEager)}
+	ASGNonRunningInstanceStates = []string{Pending, PendingWait, PendingProceed, Terminating, TerminatingWait, TerminatingProceed}
+)
+
+//ASG Instance states
+const (
+	InService          = "InService"
+	Pending            = "Pending"
+	PendingWait        = "Pending:Wait"
+	PendingProceed     = "Pending:Proceed"
+	Terminating        = "Terminating"
+	TerminatingWait    = "Terminating:Wait"
+	TerminatingProceed = "Terminating:Proceed"
```

Comment:
I don't think we need to define this, these should be included in the autoscaling pkg as constants..
see autoscaling.LifecycleState<> constants

Reply:
Found it. I shall use this one.
https://github.com/aws/aws-sdk-go/blob/4a3fa399d8948450087b360a1d82e8298eb1c25c/service/autoscaling/api.go#L14900